drizzle/migrations/0001_right_tiger_shark.sql

```diff
@@ -0,0 +1,29 @@
+CREATE TABLE "categories" (
+	"id" serial PRIMARY KEY NOT NULL,
+	"name" text NOT NULL,
+	"field_id" integer NOT NULL
+);
+--> statement-breakpoint
+CREATE TABLE "fields" (
+	"id" serial PRIMARY KEY NOT NULL,
+	"name" text NOT NULL
+);
+--> statement-breakpoint
+ALTER TABLE "papers" RENAME COLUMN "content" TO "notes";--> statement-breakpoint
+ALTER TABLE "papers" ADD COLUMN "field_id" integer NOT NULL;--> statement-breakpoint
+ALTER TABLE "papers" ADD COLUMN "category_id" integer NOT NULL;--> statement-breakpoint
+ALTER TABLE "papers" ADD COLUMN "keywords" jsonb NOT NULL;--> statement-breakpoint
+ALTER TABLE "papers" ADD COLUMN "ipfs_cid" text NOT NULL;--> statement-breakpoint
+ALTER TABLE "papers" ADD COLUMN "ipfs_url" text NOT NULL;--> statement-breakpoint
+ALTER TABLE "categories" ADD CONSTRAINT "categories_field_id_fields_id_fk" FOREIGN KEY ("field_id") REFERENCES "public"."fields"("id") ON DELETE cascade ON UPDATE no action;--> statement-breakpoint
+ALTER TABLE "papers" ADD CONSTRAINT "papers_field_id_fields_id_fk" FOREIGN KEY ("field_id") REFERENCES "public"."fields"("id") ON DELETE no action ON UPDATE no action;--> statement-breakpoint
+ALTER TABLE "papers" ADD CONSTRAINT "papers_category_id_categories_id_fk" FOREIGN KEY ("category_id") REFERENCES "public"."categories"("id") ON DELETE no action ON UPDATE no action;--> statement-breakpoint
+CREATE INDEX "title_idx" ON "papers" USING btree ("title");--> statement-breakpoint
+CREATE INDEX "abstract_idx" ON "papers" USING btree ("abstract");--> statement-breakpoint
+CREATE INDEX "field_id_idx" ON "papers" USING btree ("field_id");--> statement-breakpoint
+CREATE INDEX "category_id_idx" ON "papers" USING btree ("category_id");--> statement-breakpoint
+CREATE INDEX "keywords_gin_idx" ON "papers" USING gin ("keywords");--> statement-breakpoint
+CREATE INDEX "search_index" ON "papers" USING gin ((
```

Comment:
Adding a composite index on the `title` and `abstract` fields for full-text search when the papers search feature is implemented.

Useful links: 
- https://www.postgresql.org/docs/current/textsearch-controls.html
- https://neon.com/postgresql/postgresql-indexes/postgresql-full-text-search